src/algebra/char_p/exp_char.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jakob Scholbach.
+-/
+import algebra.char_p.basic
+import algebra.char_zero
+import data.nat.prime
+
+/-!
+# Exponential characteristic
+
+This file defines the exponential characteristic of an integral domain, 
+and establishes a few basic facts relating the exponential characteristic to the (ordinary)
+characteristic.
+
+## Main results
+- `exp_char` is the definition  
+- `exp_char_is_prime_or_one` the exponential characteristic is a prime or one
+- `char_eq_exp_char_iff` the characteristic equals the exponential characteristic iff the 
+characteristic is prime
+
+## Tags
+exponential characteristic, characteristic
+-/
+
+universe u
+variables (R : Type u) [integral_domain R]
+
+/-- The definition of the exponential characteristic of an integral domain. -/
+class exp_char (q : ℕ) : Prop :=
+  (exp_char_def : (q = 1 ∧ char_zero R) ∨ (q.prime ∧ char_p R q))
+
+/-- The exponential characteristic is one if the characteristic is zero. -/
+lemma exp_char_one_of_char_zero (p q : ℕ) [hp: char_p R p] [hq : exp_char R q] (p0 : p = 0) :
```

Comment:
Any reason not to state this without the `p`?
```suggestion
/-- The exponential characteristic is one if the characteristic is zero. -/
lemma exp_char_one_of_char_zero (q : ℕ) [hp: char_p R 0] [hq : exp_char R q] :
```

Reply:
Yes, this was a silly leftover, I am sorry. I have fixed this and also similarly for the `char_zero_of_exp_char_one`.